server/forms.py

```diff
@@ -419,6 +430,21 @@ class EditClientForm(ClientForm):
         description="Do not fill out or render.",
         validators=[validators.optional()])
 
+    def __init__(self, obj=None, **kwargs):
+        self.obj = obj
+        super(ClientForm, self).__init__(obj=obj, **kwargs)
+
+    def validate(self):
+        # if our validators do not pass
+        if not super(ClientForm, self).validate():
```

Comment:
No this needs to be declared in EditClientForm as it has different functionality (checks to see if the `client_id` is changed before checking the query).